filelog.go

```diff
@@ -136,28 +163,42 @@ func (w *FileLogWriter) Rotate() {
 	w.rot <- true
 }
 
+// Generate the next filename for rotation
+func (w *FileLogWriter) nextFilename(filename string, suffix string) (string, error) {
+        // Attempt filename.suffix
+	fullName := fmt.Sprintf("%s.%s", filename, suffix)
+	if _, err := os.Stat(fullName); os.IsNotExist(err) {
+		// File does not exist, return it as the next filename
+		return fullName, nil
+	}
+
+	// If necessary, add integer suffix
+	for i := 1; i < 10000; i++ {
+		fullNameWithSuffix := fmt.Sprintf("%s.%s.%04d", filename, suffix, i)
+		if _, err := os.Stat(fullNameWithSuffix); os.IsNotExist(err) {
```

Comment:
@alanctgardner You're right about `os.IsNotExist` - I wrote a quick test program to verify. Also, I updated the wording to be more clear and included the last `Stat` error.